File: Student Info/Python/studentInfo.py
import Student

studentList = []

def run():
    quit = False
    while quit == False:
        displayOptions()
        quit = handler()
    

def displayOptions():
    print("Welcome to the student dossier!")
    print("What would you like to do?")
    print("1) Add Student")
    print("2) Remove Student")
    print("3) List all students")
    print("4) Quit")
    
def handler():
    option = int(input("Select option: "))
    print()
    if option == 1:
        addStudent()
    elif option == 2:
        removeStudent()
    elif option == 3:
        listStudents()
    elif option == 4:
        return True
    print()
    return False
    
def listStudents():
    print("List of students:")
    print("Number of students: {}\n".format(len(studentList)))
    if len(studentList)== 0:
        print("No students yet!")
    else:    
        for student in studentList:
            print("Name: {}".format(student.name))
            print("Age: {}".format(student.age))
            print("GPA: {}".format(student.gpa))
            print("Grade Level: {}".format(student.gradeLevel))
    
def addStudent():
    studentList.append(createStudent())
    print("Student Added!")
    
def removeStudent():
    if len(studentList) == 0:
        print("No students yet!")
        return
    
    found = False
    name = input("Input name of student to remove: ")
    for student in studentList:
        if student.name == name:
            studentList.remove(student)
            print("Student Removed!")
            found = True
    
    if found == False:
        print("Student not found! Remove process cancelled")
        
        
def createStudent():
    print("New Student!")
    name = input("Enter name: ")
    age = int(input("Enter age: "))
    gpa = float(input("Enter gpa: "))
    gradeLevel = input("Enter Grade Level: ")
    return Student.Student(name, age, gpa, gradeLevel)

File: Student Info/Javascript/studentinfo.js
var studentList = [];
studentDossier();

function studentDossier(){
    run();
}

function run(){
    let answer = 0;
    let quit = false;
    while((isNaN(answer) || answer >= 1 || answer <= 4) && quit == false){
        answer = 0;
        answer = parseInt(prompt("Insert Choice"));
        quit = choiceHandler(answer);
    }
}

// Returns true if 4 is input
// Else returns false
function choiceHandler(answer){
    switch(answer){
        case 1:
            addStudent();
            break;
        case 2:
            removeStudent();
            break;
        case 3:
            listAllStudents();
            break;
        case 4:
            return true;
        default:
    }
    return false;
}

function addStudent(){
    let student = {
        name : prompt("New student!\nInput name: "),
        age : getAge(),
        gpa : getGpa(),
        gradeLevel : getGradeLevel()
    };
    studentList.push(student);
}

function removeStudent(){
    let found = false;
    let name = prompt("Remove Student!\nInput name of student to remove: ");
    for(let i = 0; i < studentList.length; i++){
        if(name == studentList[i].name){
            studentList.splice(i, 1);
            found = true;
        }
    }
    if(!found){
        console.log("No such student exists!");
    } else {
        console.log(name + " was removed");
    }
}

function listAllStudents(){
    let list = 'Student List:\n';
    for(let student of studentList){
        list += `Name: ${student.name}\nAge: ${student.age}\nGPA: ${student.gpa}
                \nGrade Level: ${student.gradeLevel}\n\n`;
    }
    console.log(list);
}

/**
 * loops prompt until user inputs int value
 */
function getAge(){
    let age;
    while(isNaN(age) || !age){
        age = parseInt(prompt("Input age: "));
    }
    return age;
}


/**
 * loops prompt until user inputs float value
 */
function getGpa(){
    let gpa;
    while(isNaN(gpa) || !gpa){
        gpa = parseFloat(prompt("Input GPA: "));
    }
    return gpa;
}

/**
 * returns string of 1 of 4 grade levels
 */
function getGradeLevel(){
    let gradeLevel;
    while(!gradeLevel || isNaN(gradeLevel) || (gradeLevel < 1 && gradeLevel > 4)){
        gradeLevel = parseInt(prompt("Input Year Level:\n1)Freshman\n2)Sophomore\n3)Junior\n4)Senior"));
    }
    switch(gradeLevel){
        case 1:
            return 'Freshman';
        case 2:
            return 'Sophomore';
        case 3:
            return 'Junior';
        case 4:
            return 'Senior';
        default:
        prompt("You're not supposed to be in here");
            return 'Error in function getGradeLevel (Input beyond 1 - 4 bounds)';
    }
}